src/battle/battle.cpp

```diff
@@ -83,65 +84,122 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
 		for (size_t i = 0; i < combatants.size(); ++i)
 		{
 			if (combatants[i].first == main_character_ptr)
 			{
+				dodge_action = false;
 				std::cout << "Main character attacks!\n";
 
-				utils::Print(
-					{"Please choose your target from the available options."});
-				int options_size = foes_ptrs.size();
-				std::vector<std::string> main_character_attack_options(options_size, "");
-				for (size_t iter = 0; iter < foes_ptrs.size(); iter++)
-				{
-					main_character_attack_options[iter] = utils::UppercasedFirstChar(
-						foes_ptrs[iter]->get_display_name());
-				}
-				user_choice = utils::GetUserConstrainedChoice(main_character_attack_options);
+				int battle_options_choice = 0;
+				std::vector<std::string> battle_options{"Attack", "Dodge", "Flee"};
 
-				MainCharacterAttack(*main_character_ptr, *(foes_ptrs[user_choice - 1]));
+				// Ask for user input.
+				utils::Print({"Please choose from the available options."});
+				battle_options_choice = utils::GetUserConstrainedChoice(battle_options);
 
-				// Eliminate (erase) foes with 0 hp
-				if (foes_ptrs[user_choice - 1]->get_hp() == 0)
+				switch (battle_options_choice)
 				{
-					std::cout << utils::UppercasedFirstChar(
-									 foes_ptrs[user_choice - 1]->get_display_name())
-							  << " has been eliminated!\n";
-					for (int j = foes_ptrs.size() - 1; j >= 0; j--)
-					{
-						if (foes_ptrs[j]->get_hp() == 0)
+					// Attack option
+					case 1:
 						{
-							foes_ptrs.erase(foes_ptrs.begin() + j);
+							utils::Print({"Please choose your target from the "
+										  "available options."});
+							int options_size = foes_ptrs.size();
+							std::vector<std::string> main_character_attack_options(options_size,
+								"");
+							for (size_t iter = 0; iter < foes_ptrs.size(); iter++)
+							{
+								main_character_attack_options[iter] = utils::UppercasedFirstChar(
+									foes_ptrs[iter]->get_display_name());
+							}
+							user_choice = utils::GetUserConstrainedChoice(
+								main_character_attack_options);
+
+							MainCharacterAttack(*main_character_ptr,
+								*(foes_ptrs[user_choice - 1]));
+
+							// Eliminate (erase) foes with 0 hp
+							if (foes_ptrs[user_choice - 1]->get_hp() == 0)
+							{
+								std::cout
+									<< utils::UppercasedFirstChar(
+										   foes_ptrs[user_choice - 1]->get_display_name())
+									<< " has been eliminated!\n";
+								for (int j = foes_ptrs.size() - 1; j >= 0; j--)
+								{
+									if (foes_ptrs[j]->get_hp() == 0)
+									{
+										foes_ptrs.erase(foes_ptrs.begin() + j);
+									}
+								}
+								for (int j = combatants.size() - 1; j >= 0; j--)
+								{
+									if (combatants[j].first->get_hp() == 0)
+									{
+										combatants.erase(combatants.begin() + j);
+									}
+								}
+							}
+							break;
 						}
-					}
-					for (int j = combatants.size() - 1; j >= 0; j--)
-					{
-						if (combatants[j].first->get_hp() == 0)
+						// Dodge option
```

Comment:
Wrong format? Same with the Flee option comment.

Reply:
why is it a wrong format??

Reply:
Extra tab inserted. Comment should be aligned with the case it is describing.

Reply:
oh